server/client.go

```diff
@@ -3238,7 +3246,8 @@ func (c *client) processInboundClientMsg(msg []byte) bool {
 		if c.opts.NoResponders {
 			if sub := c.subForReply(c.pa.reply); sub != nil {
 				proto := fmt.Sprintf("HMSG %s %s 16 16\r\nNATS/1.0 503\r\n\r\n\r\n", c.pa.reply, sub.sid)
-				c.sendProtoNow([]byte(proto))
+				c.queueOutbound([]byte(proto))
+				c.addToPCD(c)
```

Comment:
These two feel like we will see them alot together, should have a helper for the two together?

Reply:
Ok, I have made a mistake in sendOK() since I used enqueueProto *and* addToPCD, which is wrong. So after fixing sendOK() - and I believe that in that case it should be enqueueProto alone, then it is actually only here that we have queueOutbound() + addToPCD() since in normal deliverMsg(), we queue several buffers and then later call addToPCD(). Now in the mqtt branch I may have another use, so I may make 1 helper but maybe later.

For sendOK(), do you have a strong opinion on queueOutbound()+addToPCD() or simple enqueueProto()? (imo, I would think the later)

Reply:
I do not have a strong opinion on either one so go with `enqueueProto`